lib/events.js

```diff
@@ -96,6 +128,38 @@ function emitOne(handler, isFn, self, arg1) {
       listeners[i].call(self, arg1);
   }
 }
+function emitOneOnce(handler, isFn, self, ev, arg1) {
+  var fired;
+  if (isFn) {
+    fired = handler.fired;
+    if (!fired)
+      handler.fired = true;
+    handler = handler.once;
+    self.removeListener(ev, handler);
+    if (fired)
+      return;
+    handler.call(self, arg1);
```

Comment:
Seems at least this part of the conditional is the exact same in `emitNoneOnce()`. There a clean way to de-duplicate that?